pilot/pkg/model/service.go

```diff
@@ -774,6 +778,21 @@ func (s *Service) GetAddressForProxy(node *Proxy) string {
 	return s.DefaultAddress
 }
 
+// GetExtraSvcAddresses returns a Service's extra addresses to the cluster where the node resides.
+// Especially for dual stack k8s service to get other IP family addresses.
+func (s *Service) GetExtraSvcAddresses(node *Proxy) []string {
+	if len(s.ExtraSvcAddresses) == 0 && node.Metadata != nil {
+		if node.Metadata.ClusterID != "" {
+			addresses := s.ClusterVIPs.GetAddressesFor(node.Metadata.ClusterID)
+			if len(addresses) > 1 {
+				s.ExtraSvcAddresses = addresses[1:]
```

Comment:
Mutating Service here isn't safe

Reply:
Yeah, based on below comment, there is no need to mutate the service here.